custom_components/easee/binary_sensor.py

```diff
@@ -0,0 +1,62 @@
+"""
+Easee charger binary sensor.
+Author: Niklas Fondberg<niklas.fondberg@gmail.com>
+"""
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+from .entity import ChargerEntity, convert_units_funcs
+from .const import (
+    DOMAIN,
+    EASEE_ENTITIES,
+)
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Setup binary sensor platform."""
+    config = hass.data[DOMAIN]["config"]
```

Comment:
We need to create a helper or some class that removes the code duplication

Reply:
Yes, it annoyed me too, just didn't see exactly how it could be done in a nice way. Could it be done in the ChargerEntity class somehow? Or perhaps do more of the heavy lifting in __init__ just before the it is forwarded to the async_forward_entry_setup.

I can give it a try, but any input on how would be appreciated :)

Reply:
I think we should have a Controller class instanciated in the __init__ file and added to the 
```
    hass.data[DOMAIN]["controller"] = new EaseeController(username, password, client_session)
```
but I need to tinker with it and think about it to see what is best. I think we can lift a lot of logic into this and keep sensor, binary_sensor, services and switch pretty clean.

Maybe you can go ahead and merge and release this until then...

Reply:
I'll re-write this to fit the new controller structure once it is merged.